sopel/tools/target.py

```diff
@@ -107,9 +108,193 @@ def add_user(self, user, privs=0):
         """
         assert isinstance(user, User)
         self.users[user.nick] = user
-        self.privileges[user.nick] = privs
+        self.privileges[user.nick] = privs or 0
         user.channels[self.name] = self
 
+    def has_privilege(self, nick, privilege):
+        """Tell if a user has a ``privilege`` level or above in this channel.
+
+        :param str nick: a user's nick in this channel
+        :param int privilege: privilege level to check
+        :rtype: bool
+
+        This method checks the user's privilege level in this channel, i.e. if
+        it has this level or higher privileges::
+
+            >>> channel.add_user(some_user, plugin.OP)
+            >>> channel.has_privilege(some_user.nick, plugin.VOICE)
+            True
+
+        The ``nick`` argument can be either a :class:`str` or a
+        :class:`sopel.tools.Identifier`. If the user is not in this channel,
+        it will be considered has not having any privilege.
+
+        .. seealso::
+
+            There are other methods to check the exact privilege level of a
+            user, such as :meth:`is_oper`, :meth:`is_owner`, :meth:`is_admin`,
+            :meth:`is_op`, :meth:`is_halfop`, and :meth:`is_voiced`.
+
+        """
+        return self.privileges.get(Identifier(nick), 0) >= privilege
+
+    def is_oper(self, nick):
+        """Tell if a user has the OPER (operator) privilege level.
+
+        :param str nick: a user's nick in this channel
+        :rtype: bool
+
+        Unlike :meth:`has_privilege`, this method checks if the user has been
+        explicitly granted the OPER privilege level::
+
+            >>> channel.add_user(some_user, plugin.OPER)
+            >>> channel.is_oper(some_user.nick)
+            True
+            >>> channel.is_voiced(some_user.nick)
+            False
+
+        Note that you can always have more than one privilege level::
+
+            >>> channel.add_user(some_user, plugin.OPER | plugin.VOICE)
+            >>> channel.is_oper(some_user.nick)
+            True
+            >>> channel.is_voiced(some_user.nick)
+            True
+
+        """
```

Comment:
This is an idea. If completely applied, this same note would also appear in the docs for `is_owner()`, `is_admin()`, `is_halfop()`, and all four relevant constants in `sopel.plugin` (`OPER`, `OWNER`, `ADMIN`, `HALFOP`).

```suggestion
        .. important::

            Not all IRC networks support this privilege mode. If you are
            writing a plugin for public distribution, ensure your code behaves
            sensibly if only ``+v`` (voice) and ``+o`` (op) modes exist.

        """
```